src/jit/codegeninterface.h

```diff
@@ -518,6 +518,12 @@ class CodeGenInterface
         siVarLoc(const LclVarDsc* varDsc, regNumber baseReg, int offset, bool isFramePointerUsed);
         siVarLoc(){};
 
+        // An overload for the inequality comparator
+        bool operator!=(const siVarLoc* rhs) const;
```

Comment:
I'm not sure it's a good idea to overload `==` and `!=` on pointers. I think a `static bool Equals(const siVarLoc*, const siVarLoc*)` would be preferable. Alternatively, `bool operator==(const siVarLoc& rhs)` which would require pointer dereferencing at the call site. However, that won't handle `nullptr`s (if that's needed).

Reply:
Personally I prefer always **const T&** over **const T**, I can change that.

I believe we are already overriding the operator== and operator!= for a lot of structures (https://github.com/dotnet/coreclr/search?q=bool+operator%3D%3D&unscoped_q=bool+operator%3D%3D). 

I am not sure if dealing with nullptr is something that hast to be considered on an Equals/operator==.

Reply:
I have discussed this with a teammate. I agree with you doing `bool Equals(const siVarLoc*, const siVarLoc*)` to deal with pointers. But I will let the operator== commented so no one add it.